dspace-api/src/main/java/org/dspace/app/orcid/service/impl/PlainMetadataSignatureGeneratorImpl.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.orcid.service.impl;
+
+import static java.util.Comparator.comparing;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang3.StringUtils;
+import org.dspace.app.orcid.service.MetadataSignatureGenerator;
+import org.dspace.content.Item;
+import org.dspace.content.MetadataValue;
+import org.dspace.core.Context;
+
+/**
+ * Implementation of {@link MetadataSignatureGenerator} that composes a
+ * signature made up of a section for each metadata value, divided by the
+ * character SIGNATURE_SECTIONS_SEPARATOR. <br/>
+ * Each section is composed of the metadata field, the metadata value and, if
+ * present, the authority, divided by the character METADATA_SECTIONS_SEPARATOR.
+ * <br/>
+ * The presence of the metadata field allows to have different signatures for
+ * metadata with the same values but referring to different fields, while the
+ * authority allows to distinguish metadata that refer to different entities,
+ * even if they have the same value. Finally, the various sections of the
+ * signature are sorted by metadata field so that the order of the input
+ * metadata values does not affect the signature.
```

Comment:
@LucaGiamminonni : Can we add some of those details about Metadata Signature to the JavaDocs?  It's important to describe usage expectations in JavaDocs, rather than just in comments on GitHub. Currently I don't see anywhere in the JavaDocs where it describes that this signature is used to check if the metadata has changed.  That's why I asked this question because the JavaDocs are vague and mostly talk about generating a signature, but don't explain *why* a signature is needed / used.